src/libraries/Microsoft.Extensions.Hosting.Abstractions/src/BackgroundService.cs

```diff
@@ -74,7 +74,17 @@ public virtual async Task StopAsync(CancellationToken cancellationToken)
             finally
             {
                 // Wait until the task completes or the stop token triggers
-                await Task.WhenAny(_executeTask, Task.Delay(Timeout.Infinite, cancellationToken)).ConfigureAwait(false);
+                try
+                {
+#if NETCOREAPP
+                    await _executeTask.WaitAsync(cancellationToken).ConfigureAwait(false);
+#else
+                    var tcs = new TaskCompletionSource<object>();
+                    using CancellationTokenRegistration registration = cancellationToken.Register(s => ((TaskCompletionSource<object>)s).SetCanceled(), tcs);
+                    await Task.WhenAny(_executeTask, tcs.Task).ConfigureAwait(false);
```

Comment:
```suggestion
                    await (await Task.WhenAny(_executeTask, tcs.Task).ConfigureAwait(false)).ConfigureAwait(false).;
```

I think since we are catching `OperationCanceledException`, we should also be awaiting the returned Task from awaiting `Task.WhenAny` here.